Add property to limit output size of a query

chliang71 opened the pull request:
This change introduces a session property to enforce limit on how much data a query can write to data sinks. This is to limit unintentionally heavy ETL queries.

```
== NO RELEASE NOTE ==
```

mayankgarg1990 reviewed (changes requested):
To be consistent with the stats - should we just use something closer to `getPhysicalWrittenDataSize` instead of `getWrittenOutputDataSize` ? That way it is easier for people to associate this metric with a real metric that they might have seen . Other than that the PR looks good

mayankgarg1990 reviewed (commented):
Actually - we should try to add a new test case for this as well - there is currently no test for this functionality

mayankgarg1990 reviewed (commented):
Lastly - this PR should have an associated release note that announces that this system property and session property now exists to control the data size that is written out.

chliang71 commented:
Thanks for the review [mayankgarg1990](https://github.com/mayankgarg1990)! Sorry for the late response, was busy with internal work. I updated the release notes and renamed the method naming to "Physical Written".

I am still looking into how to add a test though, will update.
